pkg/probe/http/http.go

```diff
@@ -58,21 +60,22 @@ func NewWithTLSConfig(config *tls.Config, followNonLocalRedirects bool) Prober {
 			DialContext: probe.ProbeDialer().DialContext,
 		})
 
-	return httpProber{transport, followNonLocalRedirects}
+	return &HttpProber{transport, followNonLocalRedirects, &sync.Map{}}
 }
 
 // Prober is an interface that defines the Probe function for doing HTTP readiness/liveness checks.
 type Prober interface {
 	Probe(req *http.Request, timeout time.Duration) (probe.Result, string, error)
 }
 
-type httpProber struct {
+type HttpProber struct {
 	transport               *http.Transport
 	followNonLocalRedirects bool
+	httpRequestCache        *sync.Map
```

Comment:
Why this is not enough and the same field is added to the `prober` struct?

Reply:
> Why this is not enough and the same field is added to the `prober` struct?

Oh, after a bunch of rebase things, I forgot the original one is still in `probe` struct.  Will delete it.